RATapi/utils/convert.py

```diff
@@ -0,0 +1,498 @@
+        "dataLimits": [project.data[c.data].data_range for c in project.contrasts],
+        "fitlowrange": [project.data[c.data].data_range[0] for c in project.contrasts],
+        "fithirange": [project.data[c.data].data_range[1] for c in project.contrasts],
+        "contrastRepeatSLDs": [[0, 1] for _ in project.contrasts],
+        "contrastFiles": [project.data[c.data].name for c in project.contrasts],
+        # some of the data is a bit complex for a (readable) list comp; create empty lists and append below
+        "contrastResolutions": [],
+        "contrastBacks": [],
+        "dataPresent": [],
+        "include_data": [],
+        "dataTypes": [],
+        "contrastLayers": [],
+        "contrastsNumberOfLayers": [],
+    }
+    for contrast in project.contrasts:
+        # R1 stores contrast resolutions and background by the index of the relevant parameter
+        resolution = project.resolutions[contrast.resolution]
+        contrasts["contrastResolutions"].append(project.resolution_parameters.index(resolution.value_1, True))
+
+        background = project.backgrounds[contrast.background]
+        contrasts["contrastBacks"].append(project.background_parameters.index(background.value_1, True))
+
+        data = project.data[contrast.data]
+        if "simulation" in data.name:
+            contrasts["dataPresent"].append(0)
+            contrasts["include_data"].append(0)
+            contrasts["dataTypes"].append("Simulation")
+        else:
+            contrasts["dataPresent"].append(1)
+            contrasts["include_data"].append(1)
+            contrasts["dataTypes"].append("Ascii File")
+
+        if project.model == LayerModels.StandardLayers:
+            model = contrast.model
+            contrasts["contrastsNumberOfLayers"].append(len(model))
+            contrasts["contrastLayers"].append(",".join(str(project.layers.index(layer) + 1) for layer in model))
+        else:
+            contrasts["contrastsNumberOfLayers"].append(0)
+            contrasts["contrastLayers"].append("")
+
+    r1.update(contrasts)
+
+    # some final processing:
+    # .mat files just contain the item for any single-item list, so process and fix that
+    # also, `savemat` will only write cells instead of matrices for object-type ndarrays. convert lists to those
+    for key, value in r1.items():
+        if isinstance(value, list):
+            if len(value) == 1:
+                r1[key] = value[0]
+            else:
+                if not all(isinstance(x, type(r1[key][0])) for x in r1[key]):
+                    r1[key] = array(value, dtype="object")
+
+    if return_struct:
+        return r1
+
+    if Path(filename).suffix != ".mat":
+        filename += ".mat"
+    # scipy.io.savemat doesn't do cells properly:
+    # https://github.com/scipy/scipy/issues/3756
+    # rather than fiddling we just use matlab
+    eng = wrappers.start_matlab().result()
```

Comment:
Please add a `try-except` here to give a better error message than "'NoneType' object has no attribute 'result'"